docs/migration-guide.md

```diff
@@ -0,0 +1,732 @@
+
+**Key Principles:**
+- **Start with Manual strategy** - Prevents automatic promotions during initial setup
+- **Migrate one worker deployment at a time** - Reduces risk and allows learning
+- **Test thoroughly in non-production** - Validate the approach before production migration
+- **Preserve running workflows** - Ensure in-flight workflows complete successfully
+- **Enable automation gradually** - Move to Progressive rollouts only after validation
+
+### Migration Phases
+
+#### Phase 1: Preparation
+1. **Install the controller** in non-production environments
+2. **Update worker code** to support versioning (if needed)
+3. **Test migration process** with non-critical workers
+4. **Prepare CI/CD pipeline changes** for new deployment method
+
+#### Phase 2: Initial Migration
+1. **Choose lowest-risk worker** to migrate first
+2. **Create `TemporalWorkerDeployment` CRD** with Manual strategy
+3. **Validate controller management** works correctly
+4. **Update deployment pipeline** for this worker
+
+#### Phase 3: Gradual Rollout
+1. **Migrate remaining workers** one at a time
+2. **Enable Progressive rollouts** for validated workers
+3. **Monitor and tune** rollout configurations
+4. **Train team** on new deployment process
+
+### Recommended Migration Order
+
+1. **Background/batch processing workers** - Lower risk, easier to validate
+2. **Internal service workers** - Limited external impact
+3. **Customer-facing workers** - Highest risk, migrate last with most care
+
+## Step-by-Step Migration
+
+### Step 1: Install the Temporal Worker Controller
+
+```bash
+# Install the controller using Helm
+helm install -n temporal-system --create-namespace \
+  temporal-worker-controller \
+  oci://docker.io/temporalio/temporal-worker-controller
+  
+```
+
+### Step 2: Create TemporalConnection Resources
+
+Define connection parameters to your Temporal server(s):
+
+```yaml
+apiVersion: temporal.io/v1alpha1
+kind: TemporalConnection
+metadata:
+  name: production-temporal
+  namespace: default
+spec:
+  hostPort: "production.abc123.tmprl.cloud:7233"
+  mutualTLSSecret: temporal-cloud-mtls  # If using mTLS
+---
+apiVersion: temporal.io/v1alpha1
+kind: TemporalConnection
+metadata:
+  name: staging-temporal
+  namespace: default
+spec:
+  hostPort: "staging.abc123.tmprl.cloud:7233"
+  mutualTLSSecret: temporal-cloud-mtls
+```
+
+### Step 3: Prepare Your Worker Code (If Needed)
+
+Most workers will work without changes, but you may need to update your worker initialization code to properly handle versioning:
+
+**Before (Unversioned):**
+```go
+// Worker connects without versioning
+worker := worker.New(client, "my-task-queue", worker.Options{})
+```
+
+**After (Versioned - Optional Enhancement):**
+```go
+// Worker can optionally use build ID from environment
+buildID := os.Getenv("WORKER_BUILD_ID")
+workerOptions := worker.Options{}
+if buildID != "" {
+    workerOptions.BuildID = buildID
+    workerOptions.UseBuildIDForVersioning = true
```

Comment:
I think this is wrong too and it's actually worker.Options.DeploymentOptions{BuildId, DeploymentName}, but double check the docs or the demo worker code